File: examples/governance_static_go/domain/activate_framework.go
package domain

import (
	"time"
	"fmt"
)

type ActivateFramework struct {
	FrameworkId string `json:"framework_id"`
	EffectiveDate time.Time `json:"effective_date"`
}

func (c ActivateFramework) CommandName() string { return "ActivateFramework" }

func (c ActivateFramework) Execute(repo RegulatoryFrameworkRepository) (*RegulatoryFramework, *ActivatedFramework, error) {
	existing, err := repo.Find(c.FrameworkId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("RegulatoryFramework not found: %s", c.FrameworkId)
	}
	existing.EffectiveDate = c.EffectiveDate
	if existing.Status != "draft" {
		return nil, nil, fmt.Errorf("cannot ActivateFramework: RegulatoryFramework is in %s state", existing.Status)
	}
	existing.Status = "active"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := ActivatedFramework{
		AggregateID: existing.ID,
		FrameworkId: c.FrameworkId,
		EffectiveDate: c.EffectiveDate,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/compliance_domain/lib/compliance_domain/regulatory_framework/commands/activate_framework.rb
module ComplianceDomain
  class RegulatoryFramework
    module Commands
      class ActivateFramework
        include Hecks::Command
        emits "ActivatedFramework"

        attr_reader :framework_id, :effective_date

        def initialize(framework_id: nil, effective_date: nil)
          @framework_id = framework_id
          @effective_date = effective_date
        end

        def call
          existing = repository.find(framework_id)
          if existing
            unless existing.status == "draft"
              raise Hecks::Error, "Cannot ActivateFramework: status must be 'draft', got '#{existing.status}'"
            end
            RegulatoryFramework.new(
              id: existing.id,
              name: existing.name,
              jurisdiction: existing.jurisdiction,
              version: existing.version,
              effective_date: effective_date,
              authority: existing.authority,
              requirements: existing.requirements,
              status: "active"
            )
          else
            raise Hecks::Error, "RegulatoryFramework not found: #{framework_id}"
          end
        end
      end
    end
  end
end
